EntityManager/src/main/java/eu/neclab/ngsildbroker/entityhandler/services/EntityService.java

```diff
@@ -1754,4 +1754,103 @@ private boolean isRemoteQueryPossible(Map<String, Object> payload, String tenant
+					remoteHost2Batch.put(remote.getItem1(), entities2Context);
+				}
+				entities2Context.add(Tuple2.of(context, remote.getItem2()));
+			}
+		}
+
+		List<Uni<List<NGSILDOperationResult>>> unis = new ArrayList<>();
+		if (!localEntities.isEmpty()) {
+			BatchRequest request = new BatchRequest(tenant, localEntities, contexts, AppConstants.MERGE_PATCH_REQUEST);
+			request.setNoOverwrite(noOverWrite);
+			if (!unis.isEmpty() && isDifferentRemoteQueryAvailable(request, remoteHost2Batch)) {
+				request.setDistributed(true);
+			} else {
+				request.setDistributed(false);
+			}
+			Uni<List<NGSILDOperationResult>> local = entityDAO.batchMergeEntity(request).onItem()
+					.transform(dbResult -> {
+						List<NGSILDOperationResult> result = Lists.newArrayList();
+						List<Map<String, Object>> successes = (List<Map<String, Object>>) dbResult.get("success");
+						List<Map<String, String>> fails = (List<Map<String, String>>) dbResult.get("failure");
+						List<Map<String, Object>> newEntities = Lists.newArrayList();
+						List<Map<String, Object>> oldEntities = Lists.newArrayList();
+						for (Map<String, Object> success : successes) {
+							String entityId = (String) success.get("id");
+							NGSILDOperationResult opResult = new NGSILDOperationResult(AppConstants.MERGE_PATCH_REQUEST,
+									entityId);
+							opResult.addSuccess(new CRUDSuccess(null, null, null, Sets.newHashSet()));
+							result.add(opResult);
+							Map<String, Object> old = (Map<String, Object>) success.get("old");
+							Map<String, Object> newEntity = (Map<String, Object>) success.get("new");
+							newEntities.add(newEntity);
+							oldEntities.add(old);
+						}
+						for (Map<String, String> fail : fails) {
+							fail.entrySet().forEach(entry -> {
+								String entityId = entry.getKey();
+								String sqlstate = entry.getValue();
+								request.removeFromPayloadAndContext(entityId);
+								NGSILDOperationResult opResult = new NGSILDOperationResult(AppConstants.MERGE_PATCH_REQUEST,
+										entityId);
+								if (sqlstate.equals(AppConstants.SQL_NOT_FOUND)) {
+									opResult.addFailure(new ResponseException(ErrorType.NotFound, entityId));
+								} else {
+									opResult.addFailure(new ResponseException(ErrorType.InvalidRequest, sqlstate));
+								}
+								result.add(opResult);
+							});
+
+						}
+						request.setBestCompleteResult(newEntities);
+						request.setPreviousEntity(oldEntities);
+						if (!request.getRequestPayload().isEmpty()) {
+							logger.debug("Append batch request sending to kafka " + request.getEntityIds());
+							MicroServiceUtils.serializeAndSplitObjectAndEmit(request, messageSize, batchEmitter,
+									objectMapper);
+						}
+						return result;
+					});
+
+			unis.add(0, local);
+		}
+		if (unis.isEmpty()) {
+			return Uni.createFrom().failure(new ResponseException(ErrorType.NotFound));
+		}
+		return Uni.combine().all().unis(unis).combinedWith(resultLists -> {
```

Comment:
Check concise format in case of merge batch api implementation.

Reply:
done